improver_tests/synthetic_data/test_generate_metadata.py

```diff
@@ -0,0 +1,407 @@
+
+    spatial_grid_values = _spatial_grid_defaults(SPATIAL_GRID_DEFAULT)
+    assert cube.coords()[0].name() == "realization"
+    assert cube.coords()[1].name() == spatial_grid_values["y"]
+    assert cube.coords()[2].name() == spatial_grid_values["x"]
+
+    for axis in ("y", "x"):
+        assert cube.coord(axis=axis).units == spatial_grid_values["units"]
+        assert cube.coord(axis=axis).coord_system == spatial_grid_values["coord_system"]
+        assert np.diff(cube.coord(axis=axis).points)[0] == pytest.approx(
+            spatial_grid_values["resolution"]
+        )
+
+    assert np.count_nonzero(cube.data) == 0
+
+    assert iris_time_to_datetime(cube.coord("time"))[0] == TIME_DEFAULT
+    assert (
+        iris_time_to_datetime(cube.coord("forecast_reference_time"))[0] == FRT_DEFAULT
+    )
+    assert cube.coord("forecast_period").points == FORECAST_PERIOD_DEFAULT
+
+
+def test_set_name_no_units():
+    """ Tests cube generated with specified name, automatically setting units, and the rest of the values set as default values """
+    name = "air_pressure"
+    cube = generate_metadata(name=name)
+
+    assert cube.standard_name == name
+    assert cube.name() == name
+    assert cube.units == "Pa"
+
+    # Assert that no other values have unexpectedly changed by returning changed values to defaults and comparing against default cube
+    default_cube = generate_metadata()
+    cube.standard_name = NAME_DEFAULT
+    cube.name = NAME_DEFAULT
+    cube.units = UNITS_DEFAULT
+
+    _check_cubes_are_same(cube, default_cube)
+
+
+def test_set_name_units():
+    """ Tests cube generated with specified name and units, and the rest of the values set as default values"""
+    name = "air_pressure"
+    units = "pascal"
+    cube = generate_metadata(name=name, units=units)
+
+    assert cube.standard_name == name
+    assert cube.name() == name
+    assert cube.units == units
+
+    # Assert that no other values have unexpectedly changed by returning changed values to defaults and comparing against default cube
+    default_cube = generate_metadata()
+    cube.standard_name = NAME_DEFAULT
+    cube.name = NAME_DEFAULT
+    cube.units = UNITS_DEFAULT
+
+    _check_cubes_are_same(cube, default_cube)
+
+
+def test_name_unknown_no_units():
+    """ Tests error raised if output variable name not in iris.std_names.STD_NAMES and no unit provided """
+    name = "temperature"
+    msg = "Units of {} are not known.".format(name)
+
+    with pytest.raises(ValueError, match=msg):
+        generate_metadata(name)
+
+
+def test_name_unknown_with_units():
+    """ Tests cube generated with specified name which isn't a CF standard name, specified units, and the rest of the values set as default values"""
+    name = "lapse_rate"
+    units = "K m-1"
+    cube = generate_metadata(name=name, units=units)
+
+    # "lapse_rate" not CF standard so standard_name expected None
+    assert cube.standard_name == None
```

Comment:
Codacy found an issue: [Comparison to None should be 'expr is None'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5894027)